File: src/Phoenix.Models/Plcs/Climatixs/Dto/ClimatixDto.cs
using Phoenix.Models.Base.Dto;

namespace Phoenix.Models.Plcs.Climatixs.Dto
{
   public sealed class ClimatixDto : PlcDtoBase
   {
      public float OutsideTemp { get; init; }
      public float ChHighInletPresure { get; init; }
      public float ChHighOutletPresure { get; init; }
      public bool Alarm { get; init; }

      public float Ch1LowInletTemp { get; init; }
      public float Ch1LowOutletTemp { get; init; }
      public float Ch1LowOutletPresure { get; init; }
      public float Ch1HeatCurveTemp { get; init; }
      public bool Ch1PumpAlarm { get; init; }
      public bool Ch1PumpStatus { get; init; }
      public byte Ch1ValvePosition { get; init; }
      public bool Ch1Status { get; init; }

      public float Ch2LowInletTemp { get; init; }
      public float Ch2LowOutletTemp { get; init; }
      public float Ch2LowOutletPresure { get; init; }
      public float Ch2HeatCurveTemp { get; init; }
      public bool Ch2PumpAlarm { get; init; }
      public bool Ch2PumpStatus { get; init; }
      public byte Ch2ValvePosition { get; init; }
      public bool Ch2Status { get; init; }

      public float DhwTemp { get; init; }
      public float DhwTempSet { get; init; }
      public bool DhwPumpAlarm { get; init; }
      public bool DhwPumpStatus { get; init; }
      public byte DhwValvePosition { get; init; }
      public bool DhwStatus { get; init; }
   }
}

File: src/Phoenix.Web/src/models/api/plcs/climatixs/dto/climatixDto.ts
import { PlcDtoBase } from '../../../base/dto/plcDtoBase';

export interface ClimatixDto extends PlcDtoBase {
   outsideTemp: number;
   chHighInletPresure: number;
   chHighOutletPresure: number;
   alarm: boolean;
   ch1LowInletTemp: number;
   ch1LowOutletTemp: number;
   ch1LowOutletPresure: number;
   ch1HeatCurveTemp: number;
   ch1PumpAlarm: boolean;
   ch1PumpStatus: boolean;
   ch1ValvePosition: number;
   ch1Status: number;
   ch2LowInletTemp: number;
   ch2LowOutletTemp: number;
   ch2LowOutletPresure: number;
   ch2HeatCurveTemp: number;
   ch2PumpAlarm: boolean;
   ch2PumpStatus: boolean;
   ch2ValvePosition: number;
   ch2Status: number;
   dhwTemp: number;
   dhwTempSet: number;
   dhwPumpAlarm: boolean;
   dhwPumpStatus: boolean;
   dhwValvePosition: number;
   dhwStatus: number;
}
